vms/avm/txs/mempool/mempool.go

```diff
@@ -199,6 +211,13 @@ func (m *mempool) RequestBuildBlock() {
 }
 
 func (m *mempool) MarkDropped(txID ids.ID, reason error) {
+	m.lock.RLock()
+	defer m.lock.RUnlock()
+
+	if _, ok := m.unissuedTxs.Get(txID); ok {
+		return
+	}
+
 	m.droppedTxIDs.Put(txID, reason)
```

Comment:
I felt like it made the most sense to enforce that a tx wasn't considered dropped if it was in the mempool. I feel like this was the original idea (notice how we evict from `droppedTxIDs` in `Add`) but we probably just forgot to have the corresponding check here as well.

Interesting, previously we would insert `errDuplicateTx` here... which is especially weird.